pkg/volume/util/operationexecutor/operation_generator_test.go

```diff
@@ -108,6 +114,149 @@ func TestOperationGenerator_GenerateUnmapVolumeFunc_PluginName(t *testing.T) {
 	}
 }
 
+func TestCancelControlPlaneExpansion(t *testing.T) {
+	tests := []struct {
+		name           string
+		recoverFeature bool
+		pvc            *v1.PersistentVolumeClaim
+		pv             *v1.PersistentVolume
+		expectedValue  bool
+	}{
+		{
+			name:           "when feature gate is disabled",
+			recoverFeature: false,
+			expectedValue:  false,
+			pv:             getFakePersistentVolume("5G"),
+			pvc:            getFakePersistentVolumeClaim("5G", "5G"),
+		},
+		{
+			name:           "feature:enabled, pvc.status bigger than pvc.size",
+			recoverFeature: true,
+			expectedValue:  true,
+			pv:             getFakePersistentVolume("10G"),
+			pvc:            getFakePersistentVolumeClaim("5G", "10G"),
+		},
+		{
+			name:           "feature:enabled, pvc.status smaller than pvc.size",
+			recoverFeature: true,
+			expectedValue:  false,
+			pv:             getFakePersistentVolume("7G"),
+			pvc:            getFakePersistentVolumeClaim("10G", "7G"),
+		},
+		{
+			name:           "feature:enabled, pv.size is smaller than pvc.size",
+			recoverFeature: true,
+			expectedValue:  false,
+			pv:             getFakePersistentVolume("7G"),
+			pvc:            getFakePersistentVolumeClaim("10G", "20G"),
```

Comment:
I wonder when can this happen where the PV capacity is 7G and PVC status is already 20G?

Reply:
For the completeness (even though it is unlikely to happen), can you also add a test case when PVC requested size < PV capacity < PVC status

Reply:
If pvc requested < pv capacity < pvc status then no volume expansion should be necessary and hence volume expansion operation should be marked as completed.

Reply:
Also added tests for scenario you mentioned though.